src/main/java/org/broadinstitute/hellbender/tools/spark/sv/evidence/XGBoostEvidenceFilter.java

```diff
@@ -0,0 +1,609 @@
+        return evidence instanceof BreakpointEvidence.ReadEvidence ?
+            ((BreakpointEvidence.ReadEvidence) evidence).getMappingQuality() : 60;
+    }
+
+    private double getTemplateSize(final BreakpointEvidence evidence) {
+        if(evidence instanceof BreakpointEvidence.ReadEvidence) {
+            // For ReadEvidence, return templateSize as percentile of library's cumulative density function:
+            final String readGroup = ((BreakpointEvidence.ReadEvidence) evidence).getReadGroup();
+            final String library = readGroupToLibraryMap.get(readGroup);
+            final LibraryStatistics libraryStatistics = libraryStatisticsMap.get(library);
+            final IntHistogram.CDF templateSizeCDF = libraryStatistics.getCDF();
+            /*
+            final IntHistogram.CDF templateSizeCDF = libraryStatisticsMap.get(
+                    ((BreakpointEvidence.ReadEvidence) evidence).getReadGroup()
+            ).getCDF();
+            */
+            final int cdfBin = Integer.min(Math.abs(((BreakpointEvidence.ReadEvidence) evidence).getTemplateSize()),
+                                           templateSizeCDF.size() - 1);
+            return templateSizeCDF.getFraction(cdfBin);
+        } else if(evidence instanceof BreakpointEvidence.TemplateSizeAnomaly) {
+            // For TemplateSizeAnomaly, return readCount scaled by coverage
+            return (double)(((BreakpointEvidence.TemplateSizeAnomaly) evidence).getReadCount()) / coverage;
+        } else {
+            throw new IllegalArgumentException(
+                    "templateSize feature only defined for ReadEvidence and TemplateSizeAnomaly"
+            );
+        }
+    }
+
+    /**
+     * For given BreakpointEvidence, calculate number of overlappers, coherent pieces of BreakpointEvidence, and
+     * the sums of their mapping qualities. Cache this information so that it can be looked up when computing max or
+     * average of these values over the set of overlapping evidence (as in getClusterOverlapInfo).
+     */
+    private void cacheOverlapInfo(final BreakpointEvidence evidence) {
+        int numOverlap = 0;
+        int overlapMappingQuality = 0;
+        int numCoherent = 0;
+        int coherentMappingQuality = 0;
+        for(final Iterator<Tuple2<BreakpointEvidence, Boolean>> overlapper_itr = evidenceOverlapChecker.overlappersWithCoherence(evidence);
+                overlapper_itr.hasNext();) {
+            final Tuple2<BreakpointEvidence, Boolean> itr_results = overlapper_itr.next();
+            final BreakpointEvidence overlapper = itr_results._1;
+            if(overlapper == evidence) {
+                continue; // don't count self-overlap
+            }
+            numOverlap += 1;
+            final int mappingQuality = getMappingQuality(overlapper);
+            overlapMappingQuality += mappingQuality;
+
+            final boolean isCoherent = itr_results._2;
```

Comment:
These underscores are one reason I suggest using a different tuple class in another comment.

Reply:
Replaced Tuple2 with ImmutablePair per your later suggestion.